packages/vc-export/src/vc-js/suites/KiltIntegritySuite.ts

```diff
@@ -0,0 +1,122 @@
+import { Crypto } from '@kiltprotocol/utils'
+import {
+  DocumentLoader,
+  ExpansionMap,
+  purposes,
+  VerificationResult,
+} from 'jsonld-signatures'
+import { JsonLdObj } from 'jsonld/jsonld-spec'
+import {
+  KILT_CREDENTIAL_DIGEST_PROOF_TYPE,
+  CredentialDigestProof,
+  DEFAULT_VERIFIABLECREDENTIAL_CONTEXT,
+} from '../../types'
+
+import { verifyCredentialDigestProof } from '../../verificationUtils'
+import KiltAbstractSuite from './KiltAbstractSuite'
+import { KILT_CREDENTIAL_CONTEXT_URL } from '../context'
+
+export default class KiltDisclosureSuite extends KiltAbstractSuite {
+  private existingProof?: CredentialDigestProof
+
+  constructor(options: { existingProof?: CredentialDigestProof } = {}) {
+    // vc-js complains when there is no verificationMethod
+    super({
+      type: KILT_CREDENTIAL_DIGEST_PROOF_TYPE,
+      verificationMethod: '<none>',
+    })
+    const { existingProof } = options
+    if (
+      existingProof &&
+      !(
+        existingProof.claimHashes instanceof Array &&
+        typeof existingProof.nonces === 'object'
+      )
+    ) {
+      throw new Error(
+        "'claimHashes' and 'nonces' are required properties on 'existingProof'"
+      )
+    }
+    this.existingProof = existingProof
+  }
+
+  public async createProof(options: {
```

Comment:
Please add a caveat in a jsdoc comment, that describes, that the vc id already has to be generated and is the same as our root hash.

Reply:
Or, that we need an `existingProof` for it to generate a new one.

Reply:
In general, I would just not offer creating proofs here, since it is not very feasible in the current state of the KILT project.
Let's keep the code in another branch for the future, when we are deciding to go deeper with vc.js.

Reply:
(same with attested `createProof` and signature `createProof`)